client/state/posts/selectors.js

```diff
@@ -29,6 +31,23 @@ export function getPost( state, globalId ) {
 }
 
 /**
+ * Returns a normalized post object by its global ID.
+ *
+ * @param  {Object} state    Global state tree
+ * @param  {String} globalId Post global ID
+ * @return {Object}          Post object
+ */
+export const getNormalizedPost = createSelector(
+	( state, globalId ) => {
+		const post = getPost( state, globalId );
+		firstPassCanonicalImage( post );
```

Comment:
Mutation of state going on here. Preferably [`cloneDeep`](https://lodash.com/docs#cloneDeep) the original, or at least `Object.assign` (or [object spread](https://github.com/sebmarkbage/ecmascript-rest-spread)).